File: src/contests/cg/puzzle/dart/rubik.dart
// 🎮 CodinGame Puzzle - rubik
// https://www.codingame.com/training/medium/rubik

import 'dart:io';

void main() {
  int n = int.parse(stdin.readLineSync()!.trim());
  // Visible mini-cubes = all cubes minus the hidden inner cube of side n-2.
  int inner = n >= 2 ? n - 2 : 0;
  print(n * n * n - inner * inner * inner);
}

File: src/contests/cg/puzzle/rust/rubik.rs
// 🎮 CodinGame Puzzle - rubik
// https://www.codingame.com/training/medium/rubik

fn rl() -> String {
    let mut s = String::new();
    std::io::stdin().read_line(&mut s).unwrap();
    s.trim().to_string()
}

fn main() {
    let n: i64 = rl().parse().unwrap();
    // Visible mini-cubes = all cubes minus the hidden inner cube of side n-2.
    let inner = if n >= 2 { n - 2 } else { 0 };
    println!("{}", n * n * n - inner * inner * inner);
}
